core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -603,6 +605,163 @@ static HttpResponse ofFailure(Throwable cause) {
+     * @param valueTypeRef the type reference for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter, TypeReference<T> valueTypeRef) {
+        return aggregateAs(httpStatusFilter, valueTypeRef, defaultSubscriberExecutor());
+    }
+
+    /**
+     * Aggregates this response. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the response are received fully.
+     * The response is converted into an instance of type {@code clazz} using the default {@link ObjectMapper}
+     * and the specified {@code httpStatusFilter} decides whether the response is to be converted or not.
+     *
+     * @param httpStatusFilter the predicate which decides whether the response is to be converted or not
+     * @param clazz the class for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter,
+                                                 Class<? extends T> clazz,
+                                                 EventExecutor executor) {
+        return aggregateAs(httpStatusFilter, (content) -> {
+            try {
+                return JacksonUtil.newDefaultObjectMapper().readValue(content, clazz);
+            } catch (JsonProcessingException e) {
+                throw new IllegalArgumentException("Unexpected exception while decoding response body: " + content, e);
+            }
+        }, executor);
+    }
+
+    /**
+     * Aggregates this response. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the response are received fully.
+     * The response is converted into an instance of type {@code valueTypeRef} using the default {@link ObjectMapper}
+     * and the specified {@code httpStatusFilter} decides whether the response is to be converted or not.
+     *
+     * @param httpStatusFilter the predicate which decides whether the response is to be converted or not
+     * @param valueTypeRef the type reference for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter,
+                                                 TypeReference<T> valueTypeRef,
+                                                 EventExecutor executor) {
+        return aggregateAs(httpStatusFilter, (content) -> {
+            try {
+                return JacksonUtil.newDefaultObjectMapper().readValue(content, valueTypeRef);
+            } catch (JsonProcessingException e) {
+                throw new IllegalArgumentException("Unexpected exception while decoding response body: " + content, e);
+            }
+        }, executor);
+    }
+
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter,
+                                                 Function<String, T> fn,
```

Comment:
HttpStatus may be useful to implement `jsonDecoder`.
```suggestion
    default <T> CompletableFuture<T> aggregateAs(Predicate<? super HttpStatus> httpStatusFilter,
                                                 Function<String, T> jsonDecoder,
```